# Chinook Music Store Business Analysis


## Background

The purpose of this project is to conduct a business analysis using SQL. **Chinook** is a fictional online music store that wants to improve their performance. Using their database, I will try to answer their questions and provide the most efficient solutions.

Below is a copy of the database schema that shows relationships between particular tables:

![Chinook database schema](https://s3.amazonaws.com/dq-content/191/chinook-schema.svg)

First, let's load the database:

In [3]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db

Let's see if it's working:

In [16]:
%%sql
SELECT * from playlists;

 * sqlite:///chinook.db
Done.


[(1, 'Music'),
 (2, 'Movies'),
 (3, 'TV Shows'),
 (4, 'Audiobooks'),
 (5, '90’s Music'),
 (6, 'Audiobooks'),
 (7, 'Movies'),
 (8, 'Music'),
 (9, 'Music Videos'),
 (10, 'TV Shows'),
 (11, 'Brazilian Music'),
 (12, 'Classical'),
 (13, 'Classical 101 - Deep Cuts'),
 (14, 'Classical 101 - Next Steps'),
 (15, 'Classical 101 - The Basics'),
 (16, 'Grunge'),
 (17, 'Heavy Metal Classic'),
 (18, 'On-The-Go 1')]

##Task 1 Selecting albums to purchase

The Chinook record store has just signed a deal with a new record label, and wants to select the first three albums that will be added to the store, from a list of four. All four albums are by artists that don't have any tracks in the store right now. Below are the artist names, and the genre of music they produce:

alt url

The record label specializes in artists from the USA, and they have given Chinook some money to advertise the new albums in the USA, so we're interested in finding out which genres sell the best in the USA.

Write a query that returns each genre, with the number of tracks sold in the USA in absolute numbers and in percentages.

In [12]:
%%sql
select g.name, count(ii.trackid) as No_of_Tracks_Sold,
round(count(*) * 100.0 / sum(count(*)) over(), 1) as Percentage_of_Tracks_Sold 
from genres g
join tracks t on
g.genreid = t.genreid
join invoice_items ii ON
t.trackid = ii.trackid
where ii.invoiceid in (SELECT
i.invoiceid
from invoices i
where billingcountry = 'USA')
group by g.name;

 * sqlite:///chinook.db
Done.


[('Alternative', 5, 1.0),
 ('Alternative & Punk', 50, 10.1),
 ('Blues', 15, 3.0),
 ('Bossa Nova', 7, 1.4),
 ('Classical', 8, 1.6),
 ('Comedy', 8, 1.6),
 ('Drama', 6, 1.2),
 ('Easy Listening', 3, 0.6),
 ('Heavy Metal', 4, 0.8),
 ('Hip Hop/Rap', 4, 0.8),
 ('Jazz', 22, 4.5),
 ('Latin', 91, 18.4),
 ('Metal', 64, 13.0),
 ('Pop', 5, 1.0),
 ('R&B/Soul', 12, 2.4),
 ('Reggae', 6, 1.2),
 ('Rock', 157, 31.8),
 ('Rock And Roll', 3, 0.6),
 ('Sci Fi & Fantasy', 5, 1.0),
 ('Science Fiction', 1, 0.2),
 ('Soundtrack', 4, 0.8),
 ('TV Shows', 14, 2.8)]

**Conclusion**

Let's look at the results above. USA based customers are the most interested in Rock music, which accounts to **31.8%** of all USA sales. The least popular genre is Science Fiction with **0.2%** of all sales.

Now, the genres of the four albums Chinook wants to add to their store are: Hip-Hop, Punk, Pop, and Blues.
From the table we can deduct, that these genres contribute to **0.8%**, **10.1%**, **1%**, and **3%** of all sales respectively. Basing on the genre sales pattern, Chinook should choose albums from these artists:
* *Red Tone* (Punk)
* *Meteor and the Girls* (Pop)
* *Slim Jim Bites* (Blues)